### ISSUE TITLE
Compiling on Ubuntu 16.04

### ISSUE THREAD

**Issue by 109C** [TheMonsterFromTheDeep/twion#3]
To compile on above release, I had to change the cflags in the makefile from
```
CFLAGS= -Wall -static-libstdc++ -static-libgcc -lglfw3 -lopengl32 -lgdi32
```
to
```
CFLAGS= -Wall -static-libstdc++ -static-libgcc -lglfw3 -lGL -lgdiplus -lX11 -lpthread -ldl
```

I don't know if there even is a solution to making it compile platform-independently, the obvious solution of changing the flags by detecting the OS sounds messy.

**Comment by TheMonsterFromTheDeep** [comment]
Yeah, sorry. I realized a while ago that this obviously couldn't be compiled cross-platform as it currently stands, because Linux and Windows do not have the same dependencies that need to be linked.

My general thought was that this project should probably be migrated to CMake, which would also improve development on Windows. However, I don't know a whole lot about CMake, and so I essentially haven't gotten around to working on that yet.

I also have future plans for moving to SDL2 instead of GLFW, which would impact the build process and dependencies as well.
